src/backend/dbus/meson.build

```diff
@@ -0,0 +1,92 @@
+if build_dbus
+  gdbus_codegen = find_program('gdbus-codegen')
+
+  unitdir = ''
+  dbus_data_dir = join_paths(get_option('prefix'), get_option('datadir'), 'dbus-1')
+  dbus_interfaces_dir = join_paths(dbus_data_dir, 'interfaces')
+  dbus_user_services_dir = join_paths(dbus_data_dir, 'services')
+  dbus_system_services_dir = join_paths(dbus_data_dir, 'system-services')
+  dbus_system_conf_dir = join_paths(dbus_data_dir, 'system.d')
+
+  px_interface = [
+    'org.libproxy.proxy.xml'
+  ]
+
+  px_interface_h = custom_target(
+    'px-interface.h',
+    input: px_interface,
+    output: ['px-interface.h'],
+    command: [
+      gdbus_codegen,
+      '--interface-info-header',
+      '--output', '@OUTPUT@',
+      '@INPUT@'
+    ]
+  )
+
+  px_interface_c = custom_target(
+    'libproxy-iface.c',
+    input: px_interface,
+    output: ['px-interface.c'],
+    command: [
+      gdbus_codegen,
+      '--interface-info-body',
+      '--output', '@OUTPUT@',
+      '@INPUT@'
+    ],
+  )
+
+  proxyd_sources = [
+    px_interface_c,
+    px_interface_h,
+    'dbus.c',
+  ]
+
+  proxyd_deps = [
+    px_backend_dep
+  ]
+
+  executable(
+    'proxyd',
+    proxyd_sources,
+    c_args: px_backend_c_args,
+    dependencies: proxyd_deps,
+    install_dir: join_paths(px_prefix, get_option('libexecdir')),
+    install: true,
+  )
+
+  # D-Bus Interface
+  install_data('org.libproxy.proxy.xml', install_dir : dbus_interfaces_dir)
+
+  # D-Bus User Service
+  user_service_data = configuration_data()
+  user_service_data.set('LIBEXECDIR', join_paths(px_prefix, get_option('libexecdir')))
+  configure_file(
+    input: 'org.libproxy.proxy.service.in',
+    output: 'org.libproxy.proxy.service',
+    configuration: user_service_data,
+    install: true,
+    install_dir: dbus_user_services_dir
+  )
+
+  # D-Bus System Service
+  system_service_data = configuration_data()
+  system_service_data.set('LIBEXECDIR', join_paths(px_prefix, get_option('libexecdir')))
+  system_service_data.set('USER', get_option('dbus-system-user'))
+  dbus_system_service = configure_file(
+    input: 'org.libproxy.proxy-system.service.in',
+    output: 'org.libproxy.proxy-system.service',
+    configuration: system_service_data
+  )
+  install_data(dbus_system_service, rename : 'org.libproxy.proxy.service', install_dir : dbus_system_services_dir)
```

Comment:
So I understand the desire to provide both system and user service, but... it seems really confusing to me.

At the very least, the system service should use a separate libproxy user account, not root.

But honestly, while it probably does make sense to expose libproxy on the session bus and while it _might_ make sense to expose it on the system bus too, perhaps we should dial back at first and just use a private D-Bus connection rather than the message buses. That way, you can make D-Bus a mandatory dependency since even Windows supports use of private connections. You get all the benefits of process separation. And you expose exactly the same API surface as the original libproxy, so D-Bus would be only an implementation detail rather than an API surface, and we can change our minds in the future. In particular, we don't wind up committing to supporting a session or system bus service forever.

I'm fairly confident that a session bus service makes sense, but much less confident in the system bus service. Sidestepping the problem for now makes sense to make.